### PR TITLE
perf: use mergeVariablesOverrides() to avoid cache breaks when variables are not passed

### PR DESCRIPTION
**Pull request by layershifter**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Fixes an issues that we have found with improved telemetry, see https://github.com/microsoft/fluentui/pull/13346#issuecomment-635210963.

This PR introduces `mergeVariablesOverrides()` which we already uses in `Attachment` to avoid creation of variables functions even if there is no overrides as that breaks caching.

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-bindings/src/utils/mergeVariablesOverrides.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { ComponentVariablesInput, mergeComponentVariables } from '@fluentui/styles';
+
+/**
+ mergeComponentVariables() is always creating a function even if the arguments are undefined we have this temporary
+ fix in place to avoid creating empty function because it is breaking caching we should either fix
+ mergeComponentVariables(), or handle this in a more generic way.
+ */
```
Comment (mnajdova) [inline packages/fluentui/react-bindings/src/utils/mergeVariablesOverrides.ts:7]: Do we still need this comment?

File: packages/fluentui/react-bindings/src/utils/mergeVariablesOverrides.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+export default function mergeVariablesOverrides(
```
Comment (mnajdova) [inline packages/fluentui/react-bindings/src/utils/mergeVariablesOverrides.ts:8]: should we add this under `@fluentui/styles` with the other merge functions?

File: packages/fluentui/react-bindings/src/utils/mergeVariablesOverrides.ts (hunk 1 above)
Comment (layershifter) [reply packages/fluentui/react-bindings/src/utils/mergeVariablesOverrides.ts:7]: I would say yes to provide a context why this is a solution

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx [hunk 3]
```diff
@@ -90,43 +92,33 @@ export const Table: React.FC<WithAsProp<TableProps>> &
...
-        overrideProps,
+        overrideProps: predefinedProps => ({
+          variables: mergeVariablesOverrides(variables, predefinedProps.variables),
+        }),
       });
     });
   };
 
   const renderHeader = () => {
-    if (!header) {
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Table/Table.tsx:112]: Should we really remove this check?

File: packages/fluentui/react-bindings/src/utils/mergeVariablesOverrides.ts (hunk 2 above)
Comment (layershifter) [reply packages/fluentui/react-bindings/src/utils/mergeVariablesOverrides.ts:8]: I don't have there any preference, however I placed it here as it's more attached to component's specifics

File: packages/fluentui/react-northstar/src/components/Table/TableRow.tsx [hunk 4]
```diff
@@ -113,13 +115,12 @@ const TableRow: React.FC<WithAsProp<TableRowProps>> & FluentComponentStaticProps
...
-    return _.map(items, (item: TableCellProps, index: number) => {
-      const overrideProps = handleVariablesOverrides(variables);
+    return _.map(items, (item: TableCellProps) => {
       return TableCell.create(item, {
-        defaultProps: () =>
-          getA11yProps('cell', {
-            ...overrideProps,
-          }),
+        defaultProps: () => getA11yProps('cell', {}),
+        overrideProps: predefinedProps => ({
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Table/TableRow.tsx:121]: 👍

File: packages/fluentui/react-northstar/src/components/Table/Table.tsx (hunk 3 above)
Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Table/Table.tsx:112]: Yup, as `.create()` call few lines below handles this:

https://github.com/microsoft/fluentui/blob/e5adec2a91f1b920d1b0a2ab5e29828f2a7f4c30/packages/fluentui/react-northstar/src/components/Table/Table.tsx#L111-L124

**Pull request merged**